### PR TITLE
helm 🎉 

### PR DESCRIPTION
**Pull request by hamham223**
### Overview
version 1:

```bash
kubectl create -f ./bigdl-kms-deployment-sgx.yaml -n mynamespace
kubectl get pods -n mynamespace
kubectl logs <pod-name> -n mynamespace
# delete
kubectl delete deployment bigdl-orca-ncf -n mynamespace
```
⬇️ 

version 2:
```bash
# install
bash ./install-bigdl-distribute-workflosw.sh
# uninstall
kubectl delete deployment bigdl-orca-ncf -n your-namespace
```
🔽 

version 3:
```bsah
# install
cd kubernetes # later we can change directory name
helm install bigdl-distribute-workflow ./

# see pod states
kubectl get pods
kubectl logs <pod>

# uninstall
helm delete bigdl-distribute-workflow
```

⬇️ 

version 4:
```bash
# install
helm upgrade --install bigdl-distribute-workflow ./

# see the logs and uninstall according to NOTES.txt
```

could you please take a 👀?  @hkvision

### Steps to go, (or have gone)

- [x] naive run-through
- [x] support user-specified envs ( such as `http_proxy`)
- [x] add "restartPolicy: OnFailure"
- [x] support mount volume using nfs
- [x] support lanuch jobs using helm
- [ ] polish job/pods names

### CODE REVIEW

File: kubernetes/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+apiVersion: v2
+name: bigdl-distribute-workflow
```
Comment (hkvision) [inline kubernetes/Chart.yaml]: bigdl-workflow

File: kubernetes/templates/NOTES.txt [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+CHART VERSION: {{ .Chart.Version }}
+APP VERSION: {{ .Chart.AppVersion }}
+
+** Please be patient while the chart is being deployed **
+
+Get the list of pods by executing:
+
+  kubectl get pods -l job-name=bigdl-orca-ncf
+
+A pod name will be return. Orca distribute workflow logs can be accessed by the following command:
```
Comment (hkvision) [inline kubernetes/templates/NOTES.txt]: BigDL workflow logs

File: kubernetes/templates/bigdl-distribute-workflow.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
+# BigDL Orca Distribute Workflow
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: bigdl-orca-ncf
```
Comment (hkvision) [inline kubernetes/templates/bigdl-distribute-workflow.yaml]: bigdl-ncf

File: kubernetes/templates/bigdl-distribute-workflow.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+spec:
+  template:
+    spec:
+      restartPolicy: OnFailure
+      containers:
+      - name: bigdl-distribute-workflow
```
Comment (hkvision) [inline kubernetes/templates/bigdl-distribute-workflow.yaml]: bigdl-workflow

File: kubernetes/templates/bigdl-distribute-workflow.yaml [hunk 5]
```diff
@@ -0,0 +1,46 @@
+# BigDL Orca Distribute Workflow
```
Comment (hkvision) [inline kubernetes/templates/bigdl-distribute-workflow.yaml]: BigDL Distributed Training/Inference Workflow

File: kubernetes/templates/bigdl-distribute-workflow.yaml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+apiVersion: batch/v1
+kind: Job
```
Comment (hkvision) [inline kubernetes/templates/bigdl-distribute-workflow.yaml]: rename the file to bigdl-workflow.yaml

File: kubernetes/templates/bigdl-distribute-workflow.yaml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        env:
+        - name: http_proxy
+          value: {{ .Values.httpProxy }}
+        - name: https_proxy
+          value: {{ .Values.httpsProxy }}
+        - name: no_proxy
+          value: {{ .Values.noProxy }}
+        command: ["/bin/bash", "-c"]
+        args:
+          - cd /output/yinchen;
```
Comment (hkvision) [inline kubernetes/templates/bigdl-distribute-workflow.yaml]: remove your name

File: kubernetes/templates/bigdl-distribute-workflow.yaml [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+            mkdir bigdl-distribute-workflow;
+            cd bigdl-distribute-workflow;
```
Comment (hkvision) [inline kubernetes/templates/bigdl-distribute-workflow.yaml]: just use workspace?

File: kubernetes/templates/bigdl-distribute-workflow.yaml [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+            echo "Download dataset";
+            wget "https://files.grouplens.org/datasets/movielens/ml-100k.zip" -O ml-100k.zip;
+            unzip -o ./ml-100k.zip;
+            echo "Install required software";
+            cd -;
+            git clone "https://github.com/intel-analytics/BigDL.git";
+            cd BigDL/python/orca/tutorial/NCF;
```
Comment (hkvision) [inline kubernetes/templates/bigdl-distribute-workflow.yaml]: put cd before running the command

File: kubernetes/templates/bigdl-distribute-workflow.yaml [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+            pip install tensorflow==2.9.0;
+            echo "Start distributed training";
+            python ./tf_train_spark_dataframe.py --dataset ml-100k --data_dir /output/yinchen/bigdl-distribute-workflow --model_dir /output/yinchen/bigdl-distribute-workflow;
```
Comment (hkvision) [inline kubernetes/templates/bigdl-distribute-workflow.yaml]: remove your name

File: kubernetes/templates/bigdl-distribute-workflow.yaml [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+            echo "Start distributed inference";
+            python ./tf_predict_spark_dataframe.py --dataset ml-100k --data_dir /output/yinchen/bigdl-distribute-workflow --model_dir /output/yinchen/bigdl-distribute-workflow;
+            echo "Orca NCF tutorial is completed successfully";
```
Comment (hkvision) [inline kubernetes/templates/bigdl-distribute-workflow.yaml]: BigDL NCF workflow has completed successfully

File: kubernetes/templates/bigdl-distribute-workflow.yaml (hunk 3 above)
Comment (hamham223) [reply kubernetes/templates/bigdl-distribute-workflow.yaml]: This is the job name, corresponding to `NOTES.txt` line 9. https://github.com/hamham223/BigDL/pull/29/files#diff-fa9d5f0c5a432abd99258bdde1949e496042eca73f8f81f55f56f457f211f158R9

Comment (hkvision) [reply kubernetes/templates/bigdl-distribute-workflow.yaml]: You need to modify notes as well.

**Pull request closed without merging**